ide/wg_Completion.ml

```diff
@@ -8,7 +8,17 @@
 (*         *     (see LICENSE file for the text of the license)         *)
 (************************************************************************)
 
-module Proposals = CString.Set
+module StringOrd =
+struct
+  type t = string
+  let compare s1 s2 =
+    (* we use first size, then usual comparison *)
+    let d = String.length s1 - String.length s2 in
+    if d <> 0 then d
+    else compare s1 s2
+end
+
+module Proposals = Set.Make(StringOrd)
```

Comment:
We could actually introduce an abstract type `CompletionCandidate.t` to avoid this happening, but likely not worth it.